src/wandbot/rag/retrieval.py

```
@@ -1,159 +1,267 @@
+        except Exception as e:
+            error_message = f"FUSION-RETRIEVAL: Issue with rerank api:\n{e}\n"
+            logger.error(error_message)
+            error_info = ErrorInfo(
+                component="reranker_api",
+                has_error=True,
+                error_message=str(e),
+                error_type=type(e).__name__,
+                stacktrace=''.join(traceback.format_exc()),
+                file_path=get_error_file_path(sys.exc_info()[2])
+            )
+            raise Exception(error_message)
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)